apps/meteor/server/modules/notifications/notifications.module.ts

```diff
@@ -105,7 +105,8 @@ export class NotificationsModule {
 			if (!canAccess) {
 				// verify if can preview messages from public channels
 				if (room.t === 'c' && this.userId) {
-					return Authorization.hasPermission(this.userId, 'preview-c-room');
+					const subscription = await Subscriptions.findOne({ rid: room._id });
```

Comment:
The `preview-c-room` is used specifically to grant access to users that are not yet subscribed. The condition here should be `subscribed || hasPermission`;

Also, if the user is subscribed to the room, `canAccess` will be true and this code won't even be executed.

Reply:
`canAccess` will also be true for logged-out users when Anonymous Read is enabled.